scripts/python-skywater-pdk/skywater_pdk/liberty.py

```diff
@@ -1135,38 +1217,55 @@ def main():
         for acorner in args.corner:
             if acorner in corners:
                 continue
-            print()
-            print("Unknown corner:", acorner)
+            msg_err()
+            msg_err("Unknown corner:", acorner)
             retcode = 1
         if retcode != 0:
             args.corner.clear()
 
     if not args.corner:
-        print()
-        print("Available corners for", lib+":")
-        for k, v in sorted(corners.items()):
-            print("  -", k, v[0].describe())
-        print()
-        return retcode
-
-    print("Generating", output_corner_type.name, "liberty timing files for", lib, "at", ", ".join(args.corner))
-    print()
+        if not args.list_targets:
+            msg_info()
+            msg_info("Available corners for", lib+":")
+            for k, v in sorted(corners.items()):
+                msg_info("  -", k, v[0].describe())
+            msg_info()
+            return retcode
+        else:
+            for k, v in sorted(corners.items()):
+                for t in TimingType:
+                    if t in v[0]:
+                        print(libdir.joinpath(top_liberty_file(lib, k, t)))
+            return retcode
+
+    if not args.gen_deps:
+        msg_info("Generating", output_corner_type.name, "liberty timing files for", lib, "at", ", ".join(args.corner))
+        msg_info()
+
     for corner in args.corner:
         input_corner_type, corner_cells = corners[corner]
         if output_corner_type not in input_corner_type:
-            print("Corner", corner, "doesn't support", output_corner_type, "(only {})".format(input_corner_type))
+            msg_err("Corner", corner, "doesn't support", output_corner_type, "(only {})".format(input_corner_type))
             return 1
 
         if output_corner_type == TimingType.basic and TimingType.ccsnoise in input_corner_type:
             input_corner_type = TimingType.ccsnoise
         else:
             input_corner_type = output_corner_type
 
-        generate(
-            libdir, lib,
-            corner, output_corner_type, input_corner_type,
-            corner_cells, args.output_directory
-        )
+        if args.gen_deps:
```

Comment:
I wonder if should duplicate the iteration and keep the args check top level (similar to how we handle `list_targets`) that would make the thing more verbose but a little bit more digest in term of nested context, wdyt?